File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n=nums.size();
        int i=0;
        while(i<n){
            int correctindex=nums[i];
            if(nums[i]<n && nums[i]!=i){
                int temp= nums[i];
                nums[i]=nums[correctindex];
                nums[correctindex]=temp;
            }
            else{ i++;}
        }

            //checking for missing number
            for(int i=0;i<n;i++){
                if(i!=nums[i]){
                    return i;
                }
            }
            return n;   
    }

};

File: 268-missing-number/missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        i = 0
        n = len(nums)

        while i < n:
            if nums[i] == i or nums[i] >= n:
                i += 1
            else:
                temp = nums[i]
                nums[i] = nums[temp]
                nums[temp] = temp

        for j in range(n):
            if nums[j] != j:
                return j

        return n
